ddoc-extraction.js

```diff
@@ -0,0 +1,92 @@
+var db = require('./db'),
+    async = require('async'),
+    _ = require('underscore');
+
+var getMasterDdoc = function(callback) {
+  db.medic.get('_design/medic', function(err, ddoc) {
+    // explitly call the callback with just these two params. Couch passes more
+    // params to the callback, and we don't want those feeding into async.waterfall
+    callback(err, ddoc);
+  });
+};
+
+var extractDdocs = function(ddoc, callback) {
+  var attachmentNames = Object.keys(ddoc._attachments).filter(function(name) {
+    // return name.startsWith('ddocs/compiled/'); // for nodejs > 0.12
+    return name.indexOf('ddocs/compiled/') === 0;
+  });
+
+  console.log('Found ' + attachmentNames.length + ' attached ddocs');
+
+  callback(null, ddoc._rev, attachmentNames);
+};
+
+var ddocNameFromAttachmentName = function(attachmentName) {
+  var designDocNameFromFilePath = /ddocs\/compiled\/(.+)\.json/;
```

Comment:
Hey @garethbowen does that help? I didn't split it out originally because I was hoping that the function name would be clear enough that you didn't need to dig into the specifics. Let me know if that name makes no sense to you etc.